conda/activate.py

```diff
@@ -291,9 +291,10 @@ def build_activate(self, env_name_or_prefix):
 
     def build_deactivate(self):
         # query environment
+        old_conda_prefix = self.environ.get('CONDA_PREFIX')
         old_conda_shlvl = int(self.environ.get('CONDA_SHLVL', 0))
-        old_conda_prefix = self.environ.get('CONDA_PREFIX', None)
-        if old_conda_shlvl <= 0 or old_conda_prefix is None:
+        if not old_conda_prefix or old_conda_shlvl < 1:
+            # no active environment, so cannot deactivate; do nothing
```

Comment:
This was done to mirror the other code blocks (for example in build_reactivate) that check for the same thing.